arrows/tests/test_resection_camera.h

```diff
@@ -47,6 +46,64 @@ test_resection_camera(
                          estimated_camera->center(), center_tolerance );
 }
 
+// ----------------------------------------------------------------------------
+// Helper to test camera resection
+template < typename... Args >
+void
+test_resection_camera(
+  landmark_map_sptr const& landmarks, feature_track_set_sptr const& tracks,
+  frame_id_t test_frame, camera_perspective_sptr const& expected_camera,
+  double center_tolerance, double rotation_tolerance,
+  Args... args )
+{
+  resection_camera algo;
+  test_algo_resection_camera( algo,
+    landmarks, tracks, test_frame, expected_camera,
+    center_tolerance, rotation_tolerance,
+    args... );
+}
+
+// ----------------------------------------------------------------------------
+// Helper to test camera resection
+template < typename... Args >
+void
+test_resection_camera_config(
+  landmark_map_sptr const& landmarks, feature_track_set_sptr const& tracks,
+  frame_id_t test_frame, camera_perspective_sptr const& expected_camera,
+  double center_tolerance, double rotation_tolerance,
+  Args... args )
+{
+  resection_camera algo;
+  auto config = algo.get_configuration();
+  config->set_value( "optimize_focal_length", true,
```

Comment:
>> The third argument is not necessary?
>
> description appears to be a required parameter

It isn't. I've written plenty of code that omits it, and there are plenty of other uses that omit it. It's only "required" when implementing `<algorithm>::get_configuration`.

Maybe you are confused because it isn't a defaulted parameter, but rather there are overloads of `set_value`?

Reply:
removed the unnecessary descriptions from config->set_value lines